tests/semantic_analyser.cpp

```diff
@@ -778,34 +772,22 @@ kprobe:f { @ = lhist(-10, -10, 10, 1); }
                ~~~~~~~~~~~~~~~~~~~~~~
 )");
   test_error("kprobe:f { $x = lhist(); }", R"(
-stdin:1:17-24: ERROR: lhist() should be directly assigned to a map
-kprobe:f { $x = lhist(); }
-                ~~~~~~~
-stdin:1:17-24: ERROR: lhist() requires 4 arguments (0 provided)
```

Comment:
A bit sad that we're losing this error message when two things are wrong but I understand that this might be just the price we have to pay for this change.

Reply:
Yeah, we just stop processing if there are errors after a given pass, since we don't necessarily want to overload the user. They'll still get this error, but only once `lhist` is used correctly.